docs/FCM_IMPLEMENTATION.md

```diff
@@ -0,0 +1,372 @@
+# FCM Implementation for MostroP2P Mobile
+
+## Overview
+
+This document outlines the implementation of Firebase Cloud Messaging (FCM) for push notifications in the MostroP2P mobile application. The implementation follows a privacy-preserving approach similar to **[MIP-05 (Marmot Push Notifications)](https://github.com/marmot-protocol/marmot/pull/18)**, where:
```

Comment:
can you add an overview of what aspects of marmot mip-05 are being implemented and why?